In [4]:
import pandas as pd


In [5]:
rides_data = pd.read_csv(r"D:\WORK\datascience\practice\Rides-Data-Analysis\rides_data_practice_with_times.csv")

In [7]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_absolute_error, mean_squared_error, r2_score
from sklearn.preprocessing import StandardScaler

inde = rides_data[['Distance_km','Duration_min']]
depe = rides_data['Fare']

inde_train, inde_test, depe_train, depe_test = train_test_split(inde, depe, test_size=0.3, random_state=42)

scalar = StandardScaler()
inde_train_stan = scalar.fit_transform(inde_train)
inde_test_stan = scalar.transform(inde_test)

model = LinearRegression()
model.fit(inde_train_stan,depe_train)

predict = model.predict(inde_test_stan)

mse = mean_squared_error(depe_test,predict)
mae = mean_absolute_error(depe_test,predict)
r2 = r2_score(depe_test,predict)

print(f'mean_absolute_error : {mae :.2f}')
print(f'mean_squared_error : {mse :.2f}')
print(f'r2_score : {r2 :.2f}')

mean_absolute_error : 10.96
mean_squared_error : 165.37
r2_score : -0.02


1.Mean Absolute error:165.37
Interpretation: On average, the model's predicted fare is off by 10.96 compared to the actual fare.
this level of error might be significant depending on the avg fare amount in the dataset.

2.Mean Square Error:165.37
Interpretation: The squared differences between the predicted and actual fares are large, suggesting large deviations in predictions for some rides.
High MSE indicates the model is struggling to capture the true relationship between the variables.

3.r2 score:-0.02
The R² score measures how well the independent variables explain the variation in the dependent variable. A score of -0.017 means the model performs worse than a simple baseline prediction that uses the mean fare for all predictions.
The model doesn't effectively capture the relationship between Distance_km, Duration_min, and Fare.

The model underperforms: The chosen features (Distance_km and Duration_min) might not be sufficient to predict the fare accurately.The relationship between the variables and the target (Fare) could be Non-linear, making linear regression a poor fit.